doc/holochain_101/src/planning_a_dapp.md

```diff
@@ -1 +1,69 @@
+   
+   how to test your application code
+
+This article will help you plan a dApp by providing practical considerations about the specifics of distributed applications in general, and Holochain dApps in particular. It has been remarked that holochain dApps require us to make a mental shift, first from applications whose data is centrally organized, and also from blockchain-based, data-centric dApps.
+
+Here we will provide a basic overview of concepts from cryptography that are central to holochains. 
+Then, we will consider the consequences of Holochain's cryptographic architecture for data permissioning, access, and security. 
+Because app data storage is distributed amongst user-participants, one must expect that data encryption and permissions are important for protecting privacy in accordance with the jurisdictions in which the app is operating.
+
+Remember that, as user-participants leave the application, they take their data with them. They also retain copies of other data that they held to support the DHT.  
+
+One must also re-think the dApp's business model such that it does not rely on a central authority's ability to whitelist access to a given resource. 
+
+## Cryptography in Holochain dApps
+Distributed systems rely more heavily on cryptographic patterns and techniques than centralized systems. 
+
+### Hashes
+
+### Signatures 
+
+### Encryption
+
+## Entry types 
+You can define particular entry types as public, private, or encrypted. Encryption is an in-built option for Holochain apps, meaning that you can automatically encrypt entry types by default.
+This is because Holochain's use of DHTs means that simply running the application allows user-participants access to all entries on the network.
+
+### Public entries
+Public entries are published to the app's DHT with an appropriate redundancy factor. 
+
+### Encrypted entries
+Like public entries, encrypted entries are published to the app's DHT with an appropriate redundancy factor.
+
+### Private entries
+Unlike public and encrypted entries, private entries are only on the source chain (device) of the publisher and are not shared out to the DHT. 
+Of course, one could always encrypt a private entry, too. 
+
+## Security - best practices
+A great way to begin offsetting the governance crises now typical of distributed systems (i.e. DAO hack) is to think in terms of protecting and enabling the community of user-participants in addition to cryptography.    
+
+In essence, one must consider how to prevent undesired access to the DHT. If membranes are not properly built in the dApps's DNA, having access to the source code also means having access to the entire network's entries via the DHT. Developers must treat the code, or at least the DNA taken as a whole, as if it's a key to the data. Note, too, that one can easily fork a Holochain dApp without disrupting the its activity, making it possible to retain the benefits of open-source code without some of the risks.
```

Comment:
👍

Reply:
`...dApp without disrupting the its activity...` ? There seems to be an error